src/_guides/language/effective-dart/design.md

```diff
@@ -939,9 +939,15 @@ pass later one. You're better off using named arguments for that.
 
 <div class="good">
 {% prettify dart %}
-new String.fromCharCodes(Iterable source, [int start = 0, int end])
-new DateTime(year, month, day, [hours, minutes, seconds, milliseconds])
-new Duration({days: 0, hours: 0, minutes: 0, seconds: 0, milliseconds: 0})
```

Comment:
I think the Duration example is useful, though switching to use `=` would be good.

It shows an alternative API where it makes more sense to use named parameters. The idea is that DateTime should be positional because it's not meaningful to have a DateTime that specifies, say, an hour without specifying the day and month too. For Duration, though, it is reasonable to omit any of the parameters, which is why they are named.